### DISCUSSION TITLE
how to add ./bin/op-batcher to op-batcher folder

### DISCUSSION THREAD

**Discussion by jvcByte** [ethereum-optimism/optimism#11819] [Projects]
trying to run  ./bin/op-batcher   --l2-eth-rpc=http://localhost:8545   --rollup-rpc=http://localhost:9545   --poll-interval=1s   --sub-safety-margin=6   --num-confirmations=1   --safe-abort-nonce-too-low-count=3   --resubmission-timeout=30s   --rpc.addr=0.0.0.0   --rpc.port=8548   --rpc.enable-admin   --max-channel-duration=25   --l1-eth-rpc=$L1_RPC_URL   --private-key=$GS_BATCHER_PRIVATE_KEY
but getting this error
bash: ./bin/op-batcher: No such file or directory
my question is, how do i add the bin folder and the op-batcher file?

**Comment by IntelligentGod** [comment]
The error bash: ./bin/op-batcher: No such file or directory means the op-batcher binary does not exist in your current folder’s bin directory. In Ethereum-Optimism (OP Stack v1.13.5), op-batcher is built from source — it’s not included by default.
You can clone OP Stack repository and build op-batcher using make command.
After building, check the bin folder.
`ls op-batcher/bin`
You should see op-batcher among other binaries.
Now you can run your command:
`./bin/op-batcher --l2-eth-rpc=http://localhost:8545 --rollup-rpc=http://localhost:9545 --poll-interval=1s ...
`
